packages/cli/src/concurrency/concurrency-control.service.ts

```diff
@@ -0,0 +1,158 @@
+import { Logger } from '@/Logger';
+import config from '@/config';
+import { Service } from 'typedi';
+import { ConcurrencyQueue } from './concurrency-queue';
+import { UnknownExecutionModeError } from '@/errors/unknown-execution-mode.error';
+import { InvalidConcurrencyCapError } from '@/errors/invalid-concurrency-cap.error';
+import { ExecutionRepository } from '@/databases/repositories/execution.repository';
+import { License } from '@/License';
+import type { WorkflowExecuteMode as ExecutionMode } from 'n8n-workflow';
+
+@Service()
+export class ConcurrencyControlService {
+	private readonly isEnabled: boolean;
+
+	private readonly productionCap: number;
```

Comment:
can we call this `capacity` instead? `Cap` seems confusing to me.

Reply:
I'll rename to `limit` for consistency with other envs, license features, etc.